File: src/Application Specific/Classes/Entity Collections/Order/OrderSupplierInvoices.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace Orders

        Public Class SupplierInvoice

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "SupplierInvoice Properties"

            Private _InvoiceID As Integer = 0
            Public ReadOnly Property InvoiceID() As Integer
                Get
                    Return _InvoiceID
                End Get
            End Property
            Public WriteOnly Property SetInvoiceID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceID", Value)
                End Set
            End Property

            Private _OrderID As Integer = 0
            Public ReadOnly Property OrderID() As Integer
                Get
                    Return _OrderID
                End Get
            End Property
            Public WriteOnly Property SetOrderID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderID", Value)
                End Set
            End Property

            Private _InvoiceReference As String
            Public ReadOnly Property InvoiceReference() As String
                Get
                    Return _InvoiceReference
                End Get
            End Property
            Public WriteOnly Property SetInvoiceReference() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_InvoiceReference", Value)
                End Set
            End Property

            Private _InvoiceDate As DateTime = Nothing
            Public ReadOnly Property InvoiceDate() As Date
                Get
                    Return _InvoiceDate
                End Get
            End Property
            Public WriteOnly Property SetInvoiceDate() As Object
                Set(ByVal Value As Object)
                    _InvoiceDate = Value
                End Set
            End Property

            Private _GoodsAmount As Double
            Public ReadOnly Property GoodsAmount() As Double
                Get
                    Return _GoodsAmount
                End Get
            End Property
            Public WriteOnly Property SetGoodsAmount() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_GoodsAmount", Value)
                End Set
            End Property

            Private _VATAmount As Double
            Public ReadOnly Property VATAmount() As Double
                Get
                    Return _VATAmount
                End Get
            End Property
            Public WriteOnly Property SetVATAmount() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_VATAmount", Value)
                End Set
            End Property

            Private _TotalAmount As Double
            Public ReadOnly Property TotalAmount() As Double
                Get
                    Return _TotalAmount
                End Get
            End Property
            Public WriteOnly Property SetTotalAmount() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TotalAmount", Value)
                End Set
            End Property

            Private _TaxCodeID As Integer
            Public ReadOnly Property TaxCodeID() As Integer
                Get
                    Return _TaxCodeID
                End Get
            End Property
            Public WriteOnly Property SetTaxCodeID() As Object
                Set(ByVal Value As Object)
                    _TaxCodeID = Value
                End Set
            End Property

            Private _NominalCode As String
            Public ReadOnly Property NominalCode() As String
                Get
                    Return _NominalCode
                End Get
            End Property
            Public WriteOnly Property SetNominalCode() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_NominalCode", Value)
                End Set
            End Property

            Private _ExtraRef As String
            Public ReadOnly Property ExtraRef() As String
                Get
                    Return _ExtraRef
                End Get
            End Property
            Public WriteOnly Property SetExtraRef() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ExtraRef", Value)
                End Set
            End Property

            Private _ReadyToSendToSage As Boolean
            Public ReadOnly Property ReadyToSendToSage() As Boolean
                Get
                    Return _ReadyToSendToSage
                End Get
            End Property
            Public WriteOnly Property SetReadyToSendToSage() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ReadyToSendToSage", Value)
                End Set
            End Property

            Private _SentToSage As Boolean
            Public ReadOnly Property SentToSage() As Boolean
                Get
                    Return _SentToSage
                End Get
            End Property
            Public WriteOnly Property SetSentToSage() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SentToSage", Value)
                End Set
            End Property

            Private _OldSystemInvoice As Boolean
            Public ReadOnly Property OldSystemInvoice() As Boolean
                Get
                    Return _OldSystemInvoice
                End Get
            End Property
            Public WriteOnly Property SetOldSystemInvoice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OldSystemInvoice", Value)
                End Set
            End Property

            Private _DateExported As DateTime = Nothing
            Public ReadOnly Property DateExported() As Date
                Get
                    Return _DateExported
                End Get
            End Property
            Public WriteOnly Property SetDateExported() As Object
                Set(ByVal Value As Object)
                    _DateExported = Value
                End Set
            End Property

            Private _KeyedBy As Integer
            Public ReadOnly Property KeyedBy() As Integer
                Get
                    Return _KeyedBy
                End Get
            End Property
            Public WriteOnly Property SetKeyedBy() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_KeyedBy", Value)
                End Set
            End Property
#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Order/OrderSupplierInvoices.cs
using System;
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Orders
    {
        public class SupplierInvoice
        {
            private DataTypeValidator _dataTypeValidator;

            public SupplierInvoice()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            
            private int _InvoiceID = 0;

            public int InvoiceID
            {
                get
                {
                    return _InvoiceID;
                }
            }

            public object SetInvoiceID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceID", value);
                }
            }

            private int _OrderID = 0;

            public int OrderID
            {
                get
                {
                    return _OrderID;
                }
            }

            public object SetOrderID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderID", value);
                }
            }

            private string _InvoiceReference;

            public string InvoiceReference
            {
                get
                {
                    return _InvoiceReference;
                }
            }

            public object SetInvoiceReference
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_InvoiceReference", value);
                }
            }

            private DateTime _InvoiceDate = default;

            public DateTime InvoiceDate
            {
                get
                {
                    return _InvoiceDate;
                }
            }

            public object SetInvoiceDate
            {
                set
                {
                    _InvoiceDate = Conversions.ToDate(value);
                }
            }

            private double _GoodsAmount;

            public double GoodsAmount
            {
                get
                {
                    return _GoodsAmount;
                }
            }

            public object SetGoodsAmount
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_GoodsAmount", value);
                }
            }

            private double _VATAmount;

            public double VATAmount
            {
                get
                {
                    return _VATAmount;
                }
            }

            public object SetVATAmount
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_VATAmount", value);
                }
            }

            private double _TotalAmount;

            public double TotalAmount
            {
                get
                {
                    return _TotalAmount;
                }
            }

            public object SetTotalAmount
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TotalAmount", value);
                }
            }

            private int _TaxCodeID;

            public int TaxCodeID
            {
                get
                {
                    return _TaxCodeID;
                }
            }

            public object SetTaxCodeID
            {
                set
                {
                    _TaxCodeID = Conversions.ToInteger(value);
                }
            }

            private string _NominalCode;

            public string NominalCode
            {
                get
                {
                    return _NominalCode;
                }
            }

            public object SetNominalCode
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_NominalCode", value);
                }
            }

            private string _ExtraRef;

            public string ExtraRef
            {
                get
                {
                    return _ExtraRef;
                }
            }

            public object SetExtraRef
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ExtraRef", value);
                }
            }

            private bool _ReadyToSendToSage;

            public bool ReadyToSendToSage
            {
                get
                {
                    return _ReadyToSendToSage;
                }
            }

            public object SetReadyToSendToSage
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ReadyToSendToSage", value);
                }
            }

            private bool _SentToSage;

            public bool SentToSage
            {
                get
                {
                    return _SentToSage;
                }
            }

            public object SetSentToSage
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SentToSage", value);
                }
            }

            private bool _OldSystemInvoice;

            public bool OldSystemInvoice
            {
                get
                {
                    return _OldSystemInvoice;
                }
            }

            public object SetOldSystemInvoice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OldSystemInvoice", value);
                }
            }

            private DateTime _DateExported = default;

            public DateTime DateExported
            {
                get
                {
                    return _DateExported;
                }
            }

            public object SetDateExported
            {
                set
                {
                    _DateExported = Conversions.ToDate(value);
                }
            }

            private int _KeyedBy;

            public int KeyedBy
            {
                get
                {
                    return _KeyedBy;
                }
            }

            public object SetKeyedBy
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_KeyedBy", value);
                }
            }
            
        }
    }
}